core/record.go

```diff
@@ -241,3 +230,27 @@ func (t *Timetrace) loadRecord(path string) (*Record, error) {
 
 	return &record, nil
 }
+
+func (t *Timetrace) editRecord(record *Record, plus string, minus string) error {
+	var dur time.Duration
+	var err error
+	if plus != "" {
+		dur, err = time.ParseDuration(plus)
+	} else {
+		dur, err = time.ParseDuration(minus)
+		dur = -dur
+	}
+	if err != nil {
+		return err
+	}
+	if record.End == nil {
+		return errors.New("record is still in progress")
+	}
```

Comment:
You even may move this to the very top of the function body, since no duration parsing is necessary in that case.